<?xml version="1.0"?>
<launch>
  <arg name="real_machine" default="false" />
  <arg name="simulation" default="false" />
  <arg name="robot_ns" default="gimbalrotor" />
  <arg name="robot_id" default="1" />
  <group ns="$(arg robot_ns)">

    <group if="$(arg real_machine)">
      <group unless="$(arg simulation)">
        <!-- fc & IMU & GPS -->
        <include file="$(find spinal)/launch/bridge.launch" >
          <arg name="mode" value="serial" />
          <arg name="serial_port" value="/dev/flight_controller" />
          <arg name="serial_baud" value="921600" />
        </include>
      </group>

      <!-- mocap -->
      <include file="$(find aerial_robot_base)/launch/external_module/mocap.launch">
	<arg name="robot_id" value="$(arg robot_id)"/>
      </include>
    </group>

    <!-- basic configuration for sensors (e.g. noise sigma) -->
    <rosparam file="$(find gimbalrotor)/config/sensors/imu/spinal.yaml" command="load" />
    <rosparam file="$(find aerial_robot_base)/config/sensors/mocap.yaml" command="load" />
  </group>

</launch>
